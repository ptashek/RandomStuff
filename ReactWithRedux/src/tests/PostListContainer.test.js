import React from 'react';
import PostListContainer from '../components/PostListContainer.react';
import renderer from 'react-test-renderer';

import { TestPostList, emptyFunction } from './TestData';

it('renders ok', () => {
  const component = renderer.create(
    <PostListContainer
      activePost={TestPostList[0].id}
      onPostClick={emptyFunction}
      posts={TestPostList}
    />
  );
  let tree = component.toJSON();

  expect(tree.children.length).toBe(1);
  expect(tree.children[0].children.length).toBe(2);

  tree.children[0].children[0].props.onClick();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
