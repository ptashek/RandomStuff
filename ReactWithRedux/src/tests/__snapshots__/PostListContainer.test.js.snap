exports[`test renders ok 1`] = `
<div>
  <div
    className="Post-Container">
    <div
      className="Post-Title Post-Title-Active"
      onClick={[Function]}>
      title
    </div>
    <div
      className="Post-Body">
      body
    </div>
  </div>
</div>
`;
